src/bindings/js.ts

```diff
@@ -1054,9 +1053,38 @@ export class JSBuilder extends ExportsWalker {
         sb.push("__lowerStaticArray(");
         this.makeLowerToMemory(valueType, sb);
         sb.push(", ");
+        if (valueType.isNumericValue) {
+          if (valueType == Type.u8) {
+            sb.push("Uint8Array");
+          } else if (valueType == Type.i8) {
+            sb.push("Int8Array");
+          } else if (valueType == Type.u16) {
+            sb.push("Uint16Array");
+          } else if (valueType == Type.i16) {
+            sb.push("Int16Array");
+          } else if (valueType == Type.u32) {
+            sb.push("Uint32Array");
+          } else if (valueType == Type.i32) {
+            sb.push("Int32Array");
+          } else if (valueType == Type.u64) {
+            sb.push("BigUint64Array");
+          } else if (valueType == Type.i64) {
+            sb.push("BigInt64Array");
+          } else if (valueType == Type.f32) {
+            sb.push("Float32Array");
+          } else if (valueType == Type.f64) {
+            sb.push("Float64Array");
+          } else {
+            // unreachable
+            assert(false);
+          }
```

Comment:
Also includes `bool` I believe, which would currently trigger the assertion

Reply:
Good catch